### PR TITLE
Some improvements to the changelog

### PR DESCRIPTION
**Pull request by sipa**
* Mention some more changes that happened between the imagined 0.1.0 "release" and 0.2.0.
* Mention that we're using semantic versioning.
* Make the titles actually link to the changes (see https://github.com/olivierlacan/keep-a-changelog/blob/main/CHANGELOG.md).
* Perhaps controversial: reduce the font size of the Added/Changed/Deprecated/... titles; I found it a bit excessive.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,14 +10,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+ - Added support for 128-bit wide multiplication on MSVC for x86_64 and arm64, giving roughly a 20% speedup on those platforms.
 
 ### Changed
  - Enabled modules schnorrsig, extrakeys and ECDH by default in `./configure`.
+ - The `secp256k1_nonce_function_rfc6979` nonce function, used by default by `secp256k1_ecdsa_sign`, now reduces the message hash modulo the group order to match the specification. This only affects improper use of ECDSA signing API.
 
 ### Deprecated
  - Deprecated context flags `SECP256K1_CONTEXT_VERIFY` and `SECP256K1_CONTEXT_SIGN`. Use `SECP256K1_CONTEXT_NONE` instead.
  - Renamed `secp256k1_context_no_precomp` to `secp256k1_context_static`.
+ - Renamed `secp256k1_schnorrsig_sign` to `secp256k1_schnorrsig_sign32`.
```
Comment (real-or-random) [inline CHANGELOG.md:24]: Should we have a standard "format" to indicate the module, e.g., a "schnorrsig module:" prefix or a "(schnorrsig module)" suffix?

**Review by real-or-random — COMMENTED** [review]
Concept ACK

I'm sure there are more API / important changes we did over the years, and it seems hard to draw the line. Maybe we should add a sentence below [0.2.0] that explains that the changes from 0.1.0 to this version are somewhat arbitrary (because 0.1.0 is not well defined)?

**Comment by real-or-random** [comment]
> I used 2021-12-25 as "date" for 0.1.0.

Ok yes, that makes sense, and I think this was the intention when the changelog was creates (but I had forgotten about it).

**Review by jonasnick — COMMENTED** [review]
Concept ACK

**Review by real-or-random — APPROVED** [review]
ACK c30b889f17e0e75b63cdfa2fe021b0d0b59c4526

**Review by jonasnick — COMMENTED** [review]
ACK c30b889f17e0e75b63cdfa2fe021b0d0b59c4526

**Pull request merged**